specification/redis/resource-manager/Microsoft.Cache/2017-02-01/redis.json

```diff
@@ -1077,6 +1309,70 @@
       },
       "description": "Redis cache access keys."
     },
+    "RedisFirewallRule": {
+      "description": "A firewall rule on a redis cache has a name, and describes a contiguous range of IP addresses permitted to connect",
+      "properties": {
+        "id": {
```

Comment:
Just want to make sure this is not a `Resource` type

Reply:
Please verify whether a 'location' property is needed in the payload to be considered a resource type. We never return a location for redisFirewallRule.
Please verify whether a resource must be tracked by ARM to be a 'Resource' type.
Please verify whether nested resources may be `Resource` type.

Reply:
`location` is not necessary for a model to be considered as resource.
We do have the concept of a ProxyResource like [here](https://github.com/Azure/azure-rest-api-specs/blob/737041200732e2d1885693e5d6c0114280e4673d/specification/sql/resource-manager/Microsoft.Sql/2014-04-01/backupLongTermRetentionVaults.json#L159) if that is what makes more sense regarding `RedisFirewallRule`

Reply:
@dsgouda it looks like ProxyResource is just an alias for Resource - is there any benefit to using it, or how would I decide which one makes sense?

Reply:
@ravbhatnagar can you take a look, just want to make sure this is a resource

Reply:
@johanste since Gaurav is out, can you take a look

Reply:
@ravbhatnagar / @johanste ping...

Reply:
@TimLovellSmith, the correct type depends on what kind of resource this is. See [the RPC addendum on designing resources](https://github.com/Azure/azure-resource-manager-rpc/blob/343b79206c316f02a9282457d734c6fded5ad3e0/v1.0/Addendum.md#designing-resources) to clarify what your resource should be modeled as.

Reply:
@johanste That doesn't fully clarify it for me. I understand that 'Nested Resource' is definitely what the firewall rules are, since they are not top level, but I don't understand whether ProxyResource would be acceptable for it, nor why you distinguish ProxyResource and Resource in the modelling (and whether that has consequences). What other practical considerations should be guiding this decision of whether to inherit Resource, ProxyResource, or as is currently the case, neither, in the model?

Reply:
From a client code generation perspective, the primary/functional concern is the eventual shape of the generated model. The introduction of the new reusable model types were intended to simplify life for service developers in that, if the resource type is a proxy resource type or a tracked resource, there is a corresponding model type with the appropriate constraints applied. For example, a proxy resource does not have to have a location. In other words, it should depend on how you have registered your resource types in ARM. 

I don't speak ARM manifests fluently - @ravbhatnagar can probably help out with that - but ideally the answer to which kind of resource you are should be represented there.

Reply:
@johanste Firewall rules do not have a location, but they are rules on the associated network cache - which is a parent resource. So there is no location property required when you create the rule... So it sounds like it is not a 'resource' resource? But should I go with proxyResource? I had a preconceived idea that 'proxy resource' is a concept describing top-level resources, not nested resources. The Addendum somewhat reinforces that through its language: "Objects should be proxy only resources if: They don't need to be Azure resources for one of the reasons above" Implying that proxy only resource is a separate category from nested resource (which is above).

Reply:
BTW I believe that because firewallRules are a nested resource, they don't have to be declared separately in the ARM manifest?

Reply:
Discussed further with @johanste and came to the conclusion it is a proxyResource! Commit coming...